src/main/java/com/skynest/tests/tests/RegistrationTest.java

```diff
@@ -14,14 +14,22 @@
 
 public class RegistrationTest extends BaseTest {
     @Test(dataProvider = "UserData")
-    void registering_new_valid_user_should_return_specified_response(String email, String password, String name, String surname, String phoneNumber, String address) throws JsonProcessingException {
-
-        RegistrationPayload registrationPayload = new RegistrationPayload(email, password, name, surname, phoneNumber, address);
-
-        Response response = given().log().all().header("Content-Type", "application/json")
-                .body(BaseTransformer.toJson(registrationPayload))
+    void registering_new_valid_user_should_return_specified_response(
```

Comment:
Instead of passing each RegistrationRequest field individually, why not simply pass the object itself? This would greatly simplify the method signature, and would also save you the pain of using those fields to construct a RegistrationRequest instance in the test itself. Something like this perhaps:

`    void registering_new_valid_user_should_return_specified_response(RegistrationRequest registrationRequest) throws JsonProcessingException {
        Response response =
                given()
                        .log().all().header("Content-Type", "application/json")
                        .body(BaseTransformer.objectToJson(registrationRequest))
...`

Then the DataProvider "getUserData" should be changed as follows (I used 2 RegistrationRequest objects arbitrarily): 
```
    public Object[][] getUserData() {
        return new Object[][] {
                new Object[] { RegistrationRequest.generateValidRegistrationRequest() },
                new Object[] { RegistrationRequest.generateValidRegistrationRequest() }
        };
```

This, of course, assumes there's a "generateValidRegistrationRequest" method that returns populated instances of RegistrationRequest class, and would be a good place to host the randomization logic